cpp/src/rolling/grouped_rolling.cu

```diff
@@ -665,6 +697,24 @@ std::unique_ptr<column> range_window_ASC(column_view const& input,
     input, preceding_column->view(), following_column->view(), min_periods, aggr, stream, mr);
 }
 
+/**
+ * @brief Comparator for numeric order-by columns, handling floating point NaN values.
+ */
+struct nan_aware_greater {
+  template <typename T, CUDF_ENABLE_IF(not cudf::is_floating_point<T>())>
+  __device__ bool operator()(T const& lhs, T const& rhs) const
+  {
+    return thrust::greater<T>{}(lhs, rhs);
+  }
+
+  template <typename T, CUDF_ENABLE_IF(cudf::is_floating_point<T>())>
+  __device__ bool operator()(T const& lhs, T const& rhs) const
+  {
+    if (std::isnan(lhs)) { return !std::isnan(rhs); }
+    return std::isnan(rhs) ? false : lhs > rhs;
```

Comment:
Should we also call thrust::greater<T>{}(lhs, rhs) here for consistency with the other implementation?

Reply:
Yes, I agree. Fixed.